src/components/OrgFile/components/Header/index.js

```diff
@@ -305,6 +308,26 @@ ${header.get('rawDescription')}`;
     // window.location.href = mailtoURI;
   }
 
+  handleAddNoteClick() {
+    let input = prompt('Enter a note to add to the header:');
+    if (input !== null) input = input.trim();
+    if (!input) return;
+    const { header } = this.props;
+
+    const dontIndent = this.props.dontIndent;
+
+    // Getting indentation, timestamp, and template string is impure, hence do it here:
+    const indentation = dontIndent ? '' : ' '.repeat(header.nestingLevel + 1);
+    const timestamp = moment().format('YYYY-MM-DD HH:MM');
+    // Generate note based on a template string (as defined in org-log-note-headings):
+    const noteText = `${indentation}- Note taken on [${timestamp}] \\\\
+${indentation}  ${input}`;
+    // TODO: wrap input; sth. like this https://codereview.stackexchange.com/a/171857
+
+    console.log(noteText);
+    this.props.org.addNote({ noteText });
```

Comment:
How can I pass the `action` to the `addNote` reducer? (And how is the `state` passed?)

In your the previous version, @munen , without `{ }`, the argument did not go through either.

Reply:
I only get `undefined` in the reducer...

Reply:
> How can I pass the action to the addNote reducer?

You don't pass `actions` to `reducers`. `actions` tie state and `reducer` together. 

>  (And how is the state passed?)

I added that bit to the wiring that I've provided: https://github.com/200ok-ch/organice/pull/423/files#diff-2b14c674e803e97fef65139e337183fbR317

In case it's not clear what an action and what a reducer does, I suggest to read the 'core principles' of Redux: https://redux.js.org/introduction/core-concepts or even do the tutorial: https://redux.js.org/basics/basic-tutorial/

On that note, we don't have to do some of the manual wiring of Redux, because we're also using react-redux which also has good documentation: https://react-redux.js.org/introduction/quick-start

Reply:
> I only get undefined in the reducer...

I apologize, I forgot to update one function signature in my last refactor before the push. I fixed it now: https://github.com/200ok-ch/organice/pull/423/commits/7ac512a9661b40bbf88a5927cb98f65e44637092